src/connection.js

```diff
@@ -717,10 +731,33 @@ class Connection extends EventEmitter {
     });
   }
 
+  // Returns false to apply backpressure.
   sendDataToTokenStreamParser(data) {
     return this.tokenStreamParser.addBuffer(data);
   }
 
+  // This is an internal method that is called from Request.pause().
+  // It has to check whether the passed Request object represents the currently
+  // active request, because the application might have called Request.pause()
+  // on an old inactive Request object.
+  pauseRequest(request) {
+    if (this.isRequestActive(request)) {
+      this.tokenStreamParser.pause();
```

Comment:
It's not only a test whether the request belongs to this connection. It could also be a request that is no longer active. This can happen because of a timeout. 

When the request has already timed out, has completed normally or has been previously canceled, it's better to ignore an additional cancel() call.